src/schema/schema_project.c

```diff
@@ -262,8 +262,9 @@ __wt_schema_project_slice(WT_SESSION_IMPL *session, WT_CURSOR **cp, const char *
             c = cp[arg];
             WT_RET(__pack_init(session, &pack, c->value_format));
             buf = &c->value;
-            p = (uint8_t *)buf->data;
-            end = p + buf->size;
+            end = p = (uint8_t *)buf->data;
+	    if (buf->size != 0)
```

Comment:
Why do we need the if here? Adding zero will do nothing.

Reply:
Yeah, that's really short-hand for:
```
if (end != NULL)
    end += buf->size;
```
The V4 sanitizer is complaining we're adding an offset to `NULL`. So, we can check if the pointer is `NULL`, or if the size if non-zero (in which case the pointer better not be `NULL`).

That said, you're right, it's cleaner the other way.

Pushed in eb8673a82.